dask/dataframe/core.py

```diff
@@ -2292,50 +2301,60 @@ def _get_return_type(arg, metadata):
         return DataFrame
 
 
-def _extract_pd(x):
+def _extract_meta(x, nonempty=False):
     """
-    Extract internal cache data (``_pd``) from dd.DataFrame / dd.Series
+    Extract internal cache data (``_meta``) from dd.DataFrame / dd.Series
     """
     if isinstance(x, _Frame):
-        return x._pd
+        return x._meta_nonempty if nonempty else x._meta
     elif isinstance(x, list):
-        return [_extract_pd(_x) for _x in x]
+        return [_extract_meta(_x, nonempty) for _x in x]
     elif isinstance(x, tuple):
-        return tuple([_extract_pd(_x) for _x in x])
+        return tuple([_extract_meta(_x, nonempty) for _x in x])
     elif isinstance(x, dict):
         res = {}
         for k in x:
-            res[k] = _extract_pd(x[k])
+            res[k] = _extract_meta(x[k], nonempty)
         return res
+    elif isinstance(x, Scalar):
+        # TODO: clean this up. This is fine for most things,
+        # but will fail if the scalar isn't numeric.
+        return 1
     else:
         return x
 
 
 def _emulate(func, *args, **kwargs):
     """
     Apply a function using args / kwargs. If arguments contain dd.DataFrame /
-    dd.Series, using internal cache (``_pd``) for calculation
+    dd.Series, using internal cache (``_meta``) for calculation
     """
-    return func(*_extract_pd(args), **_extract_pd(kwargs))
+    return func(*_extract_meta(args, True), **_extract_meta(kwargs, True))
 
 
-def map_partitions(func, metadata, *args, **kwargs):
+def map_partitions(func, meta, *args, **kwargs):
```

Comment:
While changing the `map_partitions` method API should we change this one as well?